server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -274,17 +298,67 @@ private Criteria createModifiesDateFilteringCriteria(Consumer consumer, Date sta
 
         criteria = criteria.createCriteria("pool")
                 .add(Restrictions.or(
-                    // Dates overlap if the start or end date is in our range
-                    Restrictions.or(
-                        Restrictions.between("startDate", startDate, endDate),
-                        Restrictions.between("endDate", startDate, endDate)),
-                    Restrictions.and(
-                        // The dates overlap if our range is completely encapsulated
+                        // Dates overlap if the start or end date is in our
+                        // range
+                        Restrictions.or(Restrictions.between("startDate",
+                                startDate, endDate),
+                                Restrictions.between("endDate", startDate, endDate)),
+                Restrictions.and(
+                        // The dates overlap if our range is completely
+                        // encapsulated
                         Restrictions.le("startDate", startDate),
                         Restrictions.ge("endDate", endDate))));
         return criteria;
     }
 
+    /**
+     * A version of list Modifying that finds Entitlements that modify
+     * input entitlements.
+     * When dealing with large amount of entitlements for which it is necessary
+     * to determine their modifier products.
+     * @param entitlement
+     * @return Entitlements that are being modified by the input entitlements
+     */
+    public Set<Entitlement> batchListModifying(List<Entitlement> entitlements) {
```

Comment:
Looks like I implemented [something very similar](https://github.com/candlepin/candlepin/pull/1073/files#diff-8864d89395315f6a5f8456d0e504fd91R365) for batch attach in #1073, with two differences:
1. I had the liberty that all the entitlements being passed to my method were of the same consumer, which is not the case here.
2. Second difference I think is an issue here. Both Master branch and batch attach explicitly exclude the entitlements being passed as input to this method from the result, which I think this implementation does not.

That being said, based on which branch is merged first, I think there there are two sets of methods in our implementation and we should remove that from one of the branches.